schemas/2015-06-01/Microsoft.Cdn.json

```diff
@@ -0,0 +1,401 @@
+            }
+          ]
+        },
+        "isCompressionEnabled": {
+          "description": "Whether or not compression is enabled",
+          "oneOf": [
+            {
+              "type": "boolean"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ]
+        },
+        "origins": {
+          "description": "The origin",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "description": "Host name of the custom domain",
+                "type": "string"
+              },
+              "properties": {
+                "$ref": "#/definitions/originCommon"
+              }
+            },
+            "required": [
+              "name",
+              "properties"
+            ]
+          }
+        }
+      },
+      "required": [
+        "origins"
+      ]
+    },
+    "endpointResourcesCommon": {
+      "type": "array",
+      "items": {
+        "allOf": [
+          {
+            "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/resourceBase"
+          },
+          {
+            "oneOf": [
+              {
+                "$ref": "http://schema.management.azure.com/schemas/2014-10-01-preview/Microsoft.Authorization.json#/resourceDefinitions/roleAssignments"
+              },
+              {
+                "$ref": "http://schema.management.azure.com/schemas/2015-01-01/Microsoft.Authorization.json#/resourceDefinitions/locks"
+              },
+              {
+                "$ref": "http://schema.management.azure.com/schemas/2015-01-01/Microsoft.Resources.json#/resourceDefinitions/links"
+              },
+              {
+                "$ref": "#/definitions/originsChild"
+              },
+              {
+                "$ref": "#/definitions/customdomainsChild"
+              }
+            ]
+          }
+        ]
+      },
+      "description": "Microsoft.Cdn/profiles/endpoints: Child resources in a CDN endpoint."
+    },
+    "originCommon": {
+      "type": "object",
+      "properties": {
+        "hostName": {
+          "description": "Host name of the origin",
+          "type": "string"
+        },
+        "httpPort": {
+          "description": "The http port number of the origin",
+          "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/numberOrExpression"
+        },
+        "httpsPort": {
+          "description": "The https port number of the origin",
+          "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/numberOrExpression"
+        }
+      },
+      "required": [
+        "hostName"
+      ]
+    },
+    "originsChild": {
```

Comment:
Do you mean originChild or originCommon? Both of them are used by others part of the schema and both of them are not directly referred outside of this file. I'm not sure if I understand this.